generator/src/lib.rs

```diff
@@ -1578,31 +1581,26 @@ pub fn derive_setters(_struct: &vkxml::Struct) -> Option<Tokens> {
         })
     });
 
-    let mut nexts = Vec::new();
     let extends_name = name_to_tokens(&format!("Extends{}", name));
-    if let Some(extends) = &_struct.extends {
-        for target in extends.split(',') {
-            let target = match target {
-                // https://github.com/KhronosGroup/Vulkan-Docs/pull/870
-                "VkPhysicalDeviceProperties" => "VkPhysicalDeviceProperties2",
-                x => x,
-            };
-            let target_ident = name_to_tokens(&format!("Extends{}", name_to_tokens(target)));
-            nexts.push(quote! {
-                unsafe impl #target_ident for #name {}
-            });
-        }
-    }
 
-    let next_function = if has_next {
+    let next_extends = _struct.extends.as_ref().map(|extends| {
+        let extends = extends
+            .split(',')
+            .find(|extend| root_create_info_names.contains(&extend.to_string()))
+            .expect("Should have a root create info");
+        name_to_tokens(&format!("Extends{}", name_to_tokens(&extends)))
+    });
+
+
+    // We only implement a next methods for root create infos
+    let next_function = if has_next && next_extends.is_none() {
         quote! {
-            pub fn next<T: #extends_name>(mut self, next: &'a mut T) -> #name_builder<'a> {
+            /// Prepend
+            pub fn push_next<T: #extends_name>(mut self, next: &'a mut T) -> #name_builder<'a> {
                 unsafe{
-                    let last_extension = ptr_chain_iter(&mut self.inner)
-                        .last()
-                        .expect("Initial ptr was null");
```

Comment:
Can we get rid of `ptr_chain_iter` too, now, and keep the API narrower?

Reply:
I wanted to use it for tests. Maybe we can just define it inside the test module.